### PR TITLE
Port from Travis-CI to GitHub Actions

### PR DESCRIPTION
**Pull request by ejona86**
Travis-CI no longer has a free tier (only a free trial). That was a
major reason we used Travis-CI, so that external contributors would be
able to run the CI on their forks. Iterating on a Travis config in a
personal repo was also quite convenient. The other reason was that
Travis-CI was safe to run even with untrusted code.

Since the introduction of the permissions field in workflows, GitHub
Actions appears safe to run untrusted code and has a free tier for
external contributors. GitHub Actions and Google Cloud Build are the
main contenders for a Kokoro replacement, but Cloud Build isn't safe for
untrusted code. Instead of migrating to Travis-CI.com from
Travis-CI.org, let's migrate to GitHub Actions and gain some familiarity.

I've really appreciated Travis-CI.org and have wanted to pay for it for
years but wasn't about to give it write permission to the repo. I'm
disappointed to migrate off it, now that the permissions issues have
been sorted out.

### CODE REVIEW

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+name: GitHub Actions Linux Testing
+
+on:
+  push:
+    branches:
+    - master
+    - 'v1.*'
+  pull_request:
+  schedule:
+  - cron: 54 19 * * SUN # weekly at a "random" time
```
Comment (sergiitk) [inline .github/workflows/testing.yml:10]:
```suggestion
  - cron: '54 19 * * SUN' # weekly at a "random" time
```

I'm not convinced quotes makes a difference in this case, but GH docs say they're needed

```yaml
on:
  schedule:
    # * is a special character in YAML so you have to quote this string
    - cron:  '30 5,17 * * *'
```

https://docs.github.com/en/actions/reference/events-that-trigger-workflows#schedule

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        java-version: ${{ matrix.jre }}
+        distribution: 'adopt'
+
+    - name: Gradle cache
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.gradle/caches
+          ~/.gradle/wrapper
+        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
```
Comment (sergiitk) [inline .github/workflows/testing.yml:36]: Do we need to have separate caches per jre versions? https://docs.gradle.org/current/userguide/common_caching_problems.html#java_version_tracking

File: .github/workflows/testing.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        restore-keys: |
+          ${{ runner.os }}-maven-
+    - name: Protobuf cache
+      uses: actions/cache@v2
+      with:
+        path: /tmp/protobuf-cache
+        key: ${{ runner.os }}-maven-${{ hashFiles('buildscripts/make_dependencies.sh') }}
+
+    - name: Build
+      run: buildscripts/kokoro/unix.sh
```
Comment (sergiitk) [inline .github/workflows/testing.yml:55]: Do we still need to set these env vars? https://github.com/grpc/grpc-java/blob/814655cdde5797854289ce5c2ec3e7b80ce0cf44/.travis.yml#L4-L8

**Review by sergiitk — APPROVED** [review]
LGTM, left a couple of questions

File: .github/workflows/testing.yml (hunk 1 above)
Comment (ejona86) [reply .github/workflows/testing.yml:10]: Looks like asterisks are for alias nodes. I only see the asterisk at the beginning of the string for those, and anyone running a cron Action every minute is probably being excessive. But yeah, let's put it in quotes.

File: .github/workflows/testing.yml (hunk 2 above)
Comment (ejona86) [reply .github/workflows/testing.yml:36]: No. That's a different type of caching. That is for "build caching" vs "artifact caching" which we are messing with here.

File: .github/workflows/testing.yml (hunk 3 above)
Comment (ejona86) [reply .github/workflows/testing.yml:55]: Yes and no. Yes, we want them, but they are actually managed by unix.sh:
https://github.com/grpc/grpc-java/blob/814655cdde5797854289ce5c2ec3e7b80ce0cf44/buildscripts/kokoro/unix.sh#L41-L46

That said, not having them set globally (and others like skipAndroid) is a bit annoying for the coveralls invocation later and is why I needed to `-x compileJava`.

**Pull request merged**